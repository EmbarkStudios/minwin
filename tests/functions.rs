mod utils;
use utils::*;

// Ensures we can find and emit arch specific functions
test!(
    arches,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            linking_style: LinkingStyle::RawDylib,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.System.Search.SQLBindCol",
        "Windows.Win32.UI.WindowsAndMessaging.GetWindowLongPtrA",
        "Windows.Win32.UI.WindowsAndMessaging.GetWindowLongPtrW",
    ]
);
