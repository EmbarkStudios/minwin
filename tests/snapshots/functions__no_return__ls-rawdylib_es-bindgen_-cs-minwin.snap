---
source: tests/functions.rs
expression: bo.bindings
---
#![allow(
    non_snake_case,
    non_upper_case_globals,
    non_camel_case_types,
    clippy::upper_case_acronyms
)]
#[link(name = "kernel32", kind = "raw-dylib")]
extern "system" {
    pub fn ExitProcess(uExitCode: u32) -> !;
    pub fn ExitThread(dwExitCode: u32) -> !;
    pub fn FatalExit(ExitCode: i32) -> !;
    pub fn FreeLibraryAndExitThread(hLibModule: HMODULE, dwExitCode: u32) -> !;
}
pub type HMODULE = isize;

